packages/server/lib/middleware/access.middleware.ts

```diff
@@ -97,16 +110,28 @@ export class AccessMiddleware {
         Result<{
             account: DBTeam;
             environment: DBEnvironment;
+            plan: DBPlan | null;
         }>
     > {
         if (!keyRegex.test(publicKey)) {
             return Err('invalid_secret_key_format');
         }
+
         const result = await environmentService.getAccountAndEnvironmentByPublicKey(publicKey);
         if (!result) {
             return Err('unknown_user_account');
         }
-        return Ok(result);
+
+        let plan: DBPlan | null = null;
+        if (flagHasPlan) {
+            const planRes = await getPlan(db.knex, { accountId: result.account.id });
+            if (planRes.isErr()) {
+                return Err('invalid_plan');
```

Comment:
would `plan_not_found` be a better error name?

Reply:
Changed